tests/numpy/test_creation_manipulation.py

```diff
@@ -78,6 +79,37 @@ def test_diag(nps_app_inst):
     np_arr = np.diag(np_arr)
     assert np.allclose(ba.get(), np_arr)
 
+def test_trace(nps_app_inst):
+    import nums.numpy as nps
+
+    assert nps_app_inst is not None
+
+    a: BlockArray = nps.array([1.0, 2.0, 3.0, 4.0])
+
+    # Test that trace doesn't work with 1-dimensional vectors
+    with pytest.raises(ValueError):
```

Comment:
This is good. Can you add another test here for dtype? Not as important (it is easy to see that it's correct), but might be worth also testing errors for arrays with 3+ axes, the offset, axis1, and axis2 params.

Reply:
Done